src/processes/poisson.jl

```diff
@@ -19,61 +24,94 @@ Base.union(p₁::PoissonProcess{<:Real}, p₂::PoissonProcess{<:Function}) = Poi
 
 Base.union(p₁::PoissonProcess{<:Function}, p₂::PoissonProcess{<:Real}) = PoissonProcess(x -> p₁.λ(x) + p₂.λ)
 
+Base.union(p₁::PoissonProcess{<:AbstractVector}, p₂::PoissonProcess{<:AbstractVector}) = PoissonProcess(x -> p₁.λ + p₂.λ)
+
 ishomogeneous(p::PoissonProcess{<:Real}) = true
 ishomogeneous(p::PoissonProcess{<:Function}) = false
-
-default_sampling_algorithm(::PoissonProcess, ::Geometry) = DiscretizedSampling()
-default_sampling_algorithm(::PoissonProcess{<:Real}, ::Box) = ProductSampling()
+ishomogeneous(p::PoissonProcess{<:Vector}) = false
+
+default_sampling_algorithm(::PoissonProcess) = DiscretizedSampling()
+default_sampling_algorithm(::PoissonProcess{<:Function}) = ThinnedSampling()
+
+default_lambda_max(p, g, algo::ThinnedSampling{T}) where {T<:Real} = algo.param
+
+function default_lambda_max(p, g, algo::ThinnedSampling{T}) where {T<:Dims}
+  # compute intensity on vertices
+  box = boundingbox(g)
+  if isnothing(algo.param)
+    m = CartesianGrid(box.min, box.max)
+  else
+    m = CartesianGrid(box.min, box.max, dims = algo.param)
+  end
+  v = vertices(m)
+
+  # obtain λmax
+  λvec = p.λ.(v)
+  λmax = maximum(λvec) + 0.05 * (maximum(λvec) - minimum(λvec))
+end
```

Comment:
Based on this implementation it seems that the lambda_max that is being computed is only a function of the Poisson process intensity lambda and the geometry/domain. We don't need to store lambda_max in a separate algo struct, at least it doesn't seem necessary.